environment.go

```diff
@@ -441,6 +460,12 @@ func (e *envState) GetSender(key SenderKey) (send.Sender, error) {
 	return sender, nil
 }
 
+func (e *envState) NotificationsQueue() amboy.Queue {
```

Comment:
I think that we don't actually need to expose this as part of the interface. 

I would set this up so that you wrap all of the senders in the sender map using 

amboy/logging.MakeQueueBackedSender() 

so that the senders themselves are buffered by the queue and then you just call Log() on senders, and let the buffering be an artifact of the configuration, which lets you write far less code in your job definition.

Reply:
done